aws/resource_aws_instance.go

```diff
@@ -113,6 +113,19 @@ func resourceAwsInstance() *schema.Resource {
 					validation.StringIsEmpty,
 					validation.IsIPv4Address,
 				),
+				ConflictsWith: []string{"private_ips"},
+			},
+
+			"private_ips": {
+				Type:     schema.TypeSet,
```

Comment:
Does this need to be `TypeList` to support the "primary" private IP being the first element? I believe the concept of the primary private IP is [important for DNS hostnames](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-instance-addressing.html#concepts-private-addresses). Not sure about the API response semantics of ordering once it is configured.

Reply:
ah yes thats more sense! i took the idea from the `network_interface` resource which uses a TypeSet as well but didn't think about the response ordering. Is it still ok to have the `private_ip` and `private_ips` conflict as they are set right now? or would be better to introduce the breaking change and use the ordering of the list to do determined primary vs. secondary?

Reply:
Ahh but then we'd have to still support the empty string in the list...hmm id side with having the 2 fields instead

Reply:
Also I'm rethinking the design here given the comment in the original issue https://github.com/terraform-providers/terraform-provider-aws/issues/7063#issuecomment-547513390 . if user is attaching interfaces via the `network_interface` blocks, should the private_ips list be applicable to these network_interfaces too? i.e. `private_ips` no longer conflicts with `network_interface` ?

Reply:
thinking about this more, in 9fc6e9c i've enabled the use of `private_ip` with the renamed `secondary_private_ips` to be more specific and not make things more confusing with IPs specified in the set vs. that provided in `private_ip`